distsql/select_result_test.go

```diff
@@ -15,38 +15,69 @@ package distsql
 
 import (
 	"context"
+	"testing"
 
-	. "github.com/pingcap/check"
 	"github.com/pingcap/tidb/kv"
 	"github.com/pingcap/tidb/sessionctx/stmtctx"
 	"github.com/pingcap/tidb/store/copr"
 	"github.com/pingcap/tidb/util/execdetails"
 	"github.com/pingcap/tidb/util/mock"
 	"github.com/pingcap/tipb/go-tipb"
+
+	"github.com/stretchr/testify/require"
 )
 
-func (s *testSuite) TestUpdateCopRuntimeStats(c *C) {
+func TestUpdateCopRuntimeStats(t *testing.T) {
```

Comment:
As they share the same `sr` variable, I don't think you should run them in sub test but directly run in `TestUpdateCopRuntimeStats`.

Reply:
I feel more like three steps in one somethings,so i put sr outside,which seems to destroy the `independence of the test`. let me fix it

> As they share the same `sr` variable, I don't think you should run them in sub test but directly run in `TestUpdateCopRuntimeStats`.

I feel more like three steps in one somethings,so i put sr outside,which seems to destroy the `independence of the test`. let me fix it

Reply:
... also pay attention on parallel. `t.Parallel` is a stmt that has its own effect, not a magic that you may or may not place it somewhere. For example, it is confused you call `t.Parallel` in both parent test and its child.

For common setup logic, see also 

https://github.com/pingcap/tidb/blob/dd5546dfd0ad8376b3c7188da7c19d8c3b1818ab/util/admin/admin_test.go#L83-L84

Reply:
thks, has already been updated